dask_expr/expr.py

```diff
@@ -439,6 +440,30 @@ def apply(self, function, *args, **kwargs):
     def replace(self, to_replace=None, value=no_default, regex=False):
         return Replace(self, to_replace=to_replace, value=value, regex=regex)
 
+    def align(self, other, join="outer", fill_value=None):
+        from dask_expr.collection import new_collection
+        from dask_expr.repartition import Repartition
+
+        dfs = [self, other]
+        if not all(df.known_divisions for df in dfs):
+            raise ValueError(
+                "Not all divisions are known, can't align "
+                "partitions. Please use `set_index` "
+                "to set the index."
+            )
+
+        divisions = list(unique(merge_sorted(*[df.divisions for df in dfs])))
+        if len(divisions) == 1:  # single value for index
+            divisions = (divisions[0], divisions[0])
+
+        left = Repartition(self, new_divisions=divisions, force=True)
+        other = Repartition(other, new_divisions=divisions, force=True)
+        aligned = _Align(left, other, join=join, fill_value=fill_value)
```

Comment:
Interesting, so this intermediate object isn't a dataframe, it's a tuple of two dataframes?  Totally fine, just interesting.  It breaks our current model a little (but in a way that's interesting to think about).

Reply:
Yes exactly. I think this is the cleanest in the sense that we don't deviate from the pandas model. But we could certainly change this at some point to merge both classes together